psl/diagnostics/src/pretty_print.rs

```diff
@@ -9,6 +17,7 @@ pub(crate) fn pretty_print(
     text: &str,
     span: Span,
     description: &str,
+    colorer: Box<impl DiagnosticColorer>,
```

Comment:
This should be `&'static dyn DiagnosticColorer` (not possible because of the generics or `impl DiagnosticColorer`, as it is you get the downside of boxing and the downside of monomorphization. My go-to would be the version with dynamic dispatch (`&'static dyn`) because it's better for compile times, but the other would work too.